packages/react-router/tests/link.test.tsx

```diff
@@ -10,35 +11,449 @@ import {
+  useLoaderData,
+  redirect,
 } from '../src'
 
 afterEach(() => {
   cleanup()
 })
 
 describe('Link', () => {
-  it('should NOT pass the "disabled" prop to the rendered Link component', async () => {
+  test('when a Link is disabled', async () => {
     const rootRoute = createRootRoute()
     const indexRoute = createRoute({
       getParentRoute: () => rootRoute,
       path: '/',
       component: () => (
-        <Link to="/" disabled>
-          Index
-        </Link>
+        <React.Fragment>
+          <h1>Index</h1>
+          <Link to="/" disabled>
+            Index
+          </Link>
+          <Link to="/posts" disabled>
+            Posts
+          </Link>
+        </React.Fragment>
+      ),
+    })
+
+    const postsRoute = createRoute({
+      getParentRoute: () => rootRoute,
+      path: '/posts',
+      component: () => (
+        <React.Fragment>
+          <h1>Posts</h1>
+          <Link to="/">Index</Link>
+          <Link to="/posts" disabled>
+            Posts
+          </Link>
+        </React.Fragment>
       ),
     })
+
     const router = createRouter({
-      routeTree: rootRoute.addChildren([indexRoute]),
+      routeTree: rootRoute.addChildren([indexRoute, postsRoute]),
       history: createMemoryHistory({ initialEntries: ['/'] }),
     })
 
     await router.load()
 
     const rendered = render(<RouterProvider router={router} />)
-    const customElement = rendered.queryByText('Index')
+    const customElement = rendered.getByRole('link', { name: 'Posts' })
+
+    expect(customElement?.hasAttribute('disabled')).toBe(false)
+    expect(customElement?.role).toBe('link')
+    expect(customElement?.getAttribute('aria-disabled')).toBe('true')
+
+    customElement.click()
+
+    await expect(
+      rendered.findByRole('header', { name: 'Posts' }),
+    ).rejects.toThrow()
+  })
+
+  test('when the root is active', async () => {
+    const rootRoute = createRootRoute()
+    const indexRoute = createRoute({
+      getParentRoute: () => rootRoute,
+      path: '/',
+      component: () => {
+        return (
+          <React.Fragment>
+            <h1>Index</h1>
+            <Link to="/" inactiveProps={{ className: 'inactive' }}>
+              Index
+            </Link>
+            <Link to="/posts" inactiveProps={{ className: 'inactive' }}>
+              Posts
+            </Link>
+          </React.Fragment>
+        )
+      },
+    })
+
+    const postsRoute = createRoute({
+      getParentRoute: () => rootRoute,
+      path: '/posts',
+      component: () => {
+        return (
+          <React.Fragment>
+            <h1>Posts</h1>
+            <Link to="/" inactiveProps={{ className: 'inactive' }}>
+              Index
+            </Link>
+          </React.Fragment>
+        )
+      },
+    })
+
+    const router = createRouter({
+      routeTree: rootRoute.addChildren([indexRoute, postsRoute]),
+      history: createMemoryHistory({ initialEntries: ['/'] }),
+    })
+
+    await router.load()
```

Comment:
For real-life testing, should we be doing the `router. load()` stuff? Because that seems like a departure from what's really happening at runtime

Reply:
How about rendering the router provider and then just waiting for something to be visible on the screen? Shouldn't the loaders be triggered automatically?

Reply:
Yeah. I just thought I can do this